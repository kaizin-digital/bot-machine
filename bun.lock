{
  "lockfileVersion": 1,
  "workspaces": {
    "": {
      "name": "bot-machine",
      "dependencies": {
        "@bot-machine/telegram-client": "^1.0.1",
        "redis": "^5.8.2",
        "zod": "^4.1.11",
      },
      "devDependencies": {
        "@types/bun": "^1.2.23",
      },
      "peerDependencies": {
        "typescript": "^5",
      },
    },
  },
  "packages": {
    "@bot-machine/telegram-client": ["@bot-machine/telegram-client@1.0.1", "", { "peerDependencies": { "typescript": "^5.0.0" } }, "sha512-M1bb8fdh0Du0uaF2GSLjeSnH0nLvAanWtsbMdM5v6LSfpVjpFG6e63/F8Dkxg41oqEuy22wtAHET5FtuSqpptw=="],

    "@redis/bloom": ["@redis/bloom@5.8.2", "", { "peerDependencies": { "@redis/client": "^5.8.2" } }, "sha512-855DR0ChetZLarblio5eM0yLwxA9Dqq50t8StXKp5bAtLT0G+rZ+eRzzqxl37sPqQKjUudSYypz55o6nNhbz0A=="],

    "@redis/client": ["@redis/client@5.8.2", "", { "dependencies": { "cluster-key-slot": "1.1.2" } }, "sha512-WtMScno3+eBpTac1Uav2zugXEoXqaU23YznwvFgkPwBQVwEHTDgOG7uEAObtZ/Nyn8SmAMbqkEubJaMOvnqdsQ=="],

    "@redis/json": ["@redis/json@5.8.2", "", { "peerDependencies": { "@redis/client": "^5.8.2" } }, "sha512-uxpVfas3I0LccBX9rIfDgJ0dBrUa3+0Gc8sEwmQQH0vHi7C1Rx1Qn8Nv1QWz5bohoeIXMICFZRcyDONvum2l/w=="],

    "@redis/search": ["@redis/search@5.8.2", "", { "peerDependencies": { "@redis/client": "^5.8.2" } }, "sha512-cNv7HlgayavCBXqPXgaS97DRPVWFznuzsAmmuemi2TMCx5scwLiP50TeZvUS06h/MG96YNPe6A0Zt57yayfxwA=="],

    "@redis/time-series": ["@redis/time-series@5.8.2", "", { "peerDependencies": { "@redis/client": "^5.8.2" } }, "sha512-g2NlHM07fK8H4k+613NBsk3y70R2JIM2dPMSkhIjl2Z17SYvaYKdusz85d7VYOrZBWtDrHV/WD2E3vGu+zni8A=="],

    "@types/bun": ["@types/bun@1.2.23", "", { "dependencies": { "bun-types": "1.2.23" } }, "sha512-le8ueOY5b6VKYf19xT3McVbXqLqmxzPXHsQT/q9JHgikJ2X22wyTW3g3ohz2ZMnp7dod6aduIiq8A14Xyimm0A=="],

    "@types/node": ["@types/node@24.5.2", "", { "dependencies": { "undici-types": "~7.12.0" } }, "sha512-FYxk1I7wPv3K2XBaoyH2cTnocQEu8AOZ60hPbsyukMPLv5/5qr7V1i8PLHdl6Zf87I+xZXFvPCXYjiTFq+YSDQ=="],

    "@types/react": ["@types/react@19.1.15", "", { "dependencies": { "csstype": "^3.0.2" } }, "sha512-+kLxJpaJzXybyDyFXYADyP1cznTO8HSuBpenGlnKOAkH4hyNINiywvXS/tGJhsrGGP/gM185RA3xpjY0Yg4erA=="],

    "bun-types": ["bun-types@1.2.23", "", { "dependencies": { "@types/node": "*" }, "peerDependencies": { "@types/react": "^19" } }, "sha512-R9f0hKAZXgFU3mlrA0YpE/fiDvwV0FT9rORApt2aQVWSuJDzZOyB5QLc0N/4HF57CS8IXJ6+L5E4W1bW6NS2Aw=="],

    "cluster-key-slot": ["cluster-key-slot@1.1.2", "", {}, "sha512-RMr0FhtfXemyinomL4hrWcYJxmX6deFdCxpJzhDttxgO1+bcCnkk+9drydLVDmAMG7NE6aN/fl4F7ucU/90gAA=="],

    "csstype": ["csstype@3.1.3", "", {}, "sha512-M1uQkMl8rQK/szD0LNhtqxIPLpimGm8sOBwU7lLnCpSbTyY3yeU1Vc7l4KT5zT4s/yOxHH5O7tIuuLOCnLADRw=="],

    "redis": ["redis@5.8.2", "", { "dependencies": { "@redis/bloom": "5.8.2", "@redis/client": "5.8.2", "@redis/json": "5.8.2", "@redis/search": "5.8.2", "@redis/time-series": "5.8.2" } }, "sha512-31vunZj07++Y1vcFGcnNWEf5jPoTkGARgfWI4+Tk55vdwHxhAvug8VEtW7Cx+/h47NuJTEg/JL77zAwC6E0OeA=="],

    "typescript": ["typescript@5.9.2", "", { "bin": { "tsc": "bin/tsc", "tsserver": "bin/tsserver" } }, "sha512-CWBzXQrc/qOkhidw1OzBTQuYRbfyxDXJMVJ1XNwUHGROVmuaeiEm3OslpZ1RV96d7SKKjZKrSJu3+t/xlw3R9A=="],

    "undici-types": ["undici-types@7.12.0", "", {}, "sha512-goOacqME2GYyOZZfb5Lgtu+1IDmAlAEu5xnD3+xTzS10hT0vzpf0SPjkXwAw9Jm+4n/mQGDP3LO8CPbYROeBfQ=="],

    "zod": ["zod@4.1.11", "", {}, "sha512-WPsqwxITS2tzx1bzhIKsEs19ABD5vmCVa4xBo2tq/SrV4RNZtfws1EnCWQXM6yh8bD08a1idvkB5MZSBiZsjwg=="],
  }
}
