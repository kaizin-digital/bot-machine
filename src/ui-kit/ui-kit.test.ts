import { describe, test, expect } from "bun:test";
import {
	WelcomeMessage,
	SuccessMessage,
	ErrorMessage,
	BulletedList,
	InfoPanel,
} from "./base.ui-kit";

describe("Base UI Kit", () => {
	test("should render a WelcomeMessage", () => {
		const result = WelcomeMessage({
			title: "Welcome to the Bot!",
			subtitle: "Let's get started.",
		});
		expect(result.text).toMatchSnapshot();
	});

	test("should render a WelcomeMessage without a subtitle", () => {
		const result = WelcomeMessage({ title: "Welcome!" });
		expect(result.text).toMatchSnapshot();
	});

	test("should render a SuccessMessage", () => {
		const result = SuccessMessage({
			title: "Order Confirmed",
			details: "Your order #123 has been processed.",
		});
		expect(result.text).toMatchSnapshot();
	});

	test("should render an ErrorMessage", () => {
		const result = ErrorMessage({
			error: "Payment Failed",
			suggestion: "Please check your card details and try again.",
		});
		expect(result.text).toMatchSnapshot();
	});

	test("should render a BulletedList", () => {
		const result = BulletedList({
			items: ["First item", "Second item", "Third item"],
		});
		expect(result.text).toMatchSnapshot();
	});

	test("should render an InfoPanel", () => {
		const result = InfoPanel({
			keyValues: {
				"Order ID": "123-XYZ",
				Status: "Shipped",
				Amount: "$99.99",
			},
		});
		expect(result.text).toMatchSnapshot();
	});
});
