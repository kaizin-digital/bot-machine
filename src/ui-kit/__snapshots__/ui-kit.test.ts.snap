// Bun Snapshot v1, https://bun.sh/docs/test/snapshots

exports[`Base UI Kit should render a WelcomeMessage 1`] = `
"🎉 <b>Welcome to the Bot!</b>

<i>Let's get started.</i>

"
`;

exports[`Base UI Kit should render a WelcomeMessage without a subtitle 1`] = `
"🎉 <b>Welcome!</b>

"
`;

exports[`Base UI Kit should render a SuccessMessage 1`] = `
"✅ <b>Order Confirmed</b>

<i>Your order #123 has been processed.</i>

"
`;

exports[`Base UI Kit should render an ErrorMessage 1`] = `
"❌ <b>Payment Failed</b>

<i>Please check your card details and try again.</i>

"
`;

exports[`Base UI Kit should render a BulletedList 1`] = `
"• First item
• Second item
• Third item
"
`;

exports[`Base UI Kit should render an InfoPanel 1`] = `
"<b>Order ID: </b><code>123-XYZ</code>
<b>Status: </b><code>Shipped</code>
<b>Amount: </b><code>$99.99</code>
"
`;
